src/LitePatch/LitePatch.Services/Services/GitPatchService.cs

```diff
@@ -1,38 +1,43 @@
 ﻿using System.Management.Automation;
+using LibGit2Sharp;
 using LitePatch.Services.Interfaces;
+using LitePatch.Services.Models;
+using LitePatch.Services.Repo;
 
 namespace LitePatch.Services.Services;
 
-public class GitPatchService(IGitInfoService gitInfoService, ISettingsService settingsService) : IGitPatchService
+public class GitPatchService() : IGitPatchService
 {
-    private int _counter = 1;
-    
-    public void CreatePatch(string sha, string commitName)
-    {
-        var output = Path.Combine(settingsService.Settings.OutputFolderPath,
-            $"{_counter.ToString().PadLeft(4, '0')}-{CleanPatchName(commitName)}.patch");
-        
-        var command = @$"git format-patch -1 -B1% {sha} --output='{output}'";
 
-        using var powershell = PowerShell.Create();
-        powershell.AddScript(@$"cd {gitInfoService.RepositoryPath}");
-        powershell.AddScript(command);
+    private readonly IPatchRepository _patchRepository;
+    private readonly ISettingsService _settingsService;
 
-        var results = powershell.Invoke();
-        _counter++;
+    public GitPatchService(IPatchRepository patchRepository, ISettingsService settingsService) : this()
+    {
+        _patchRepository = patchRepository;
+        var outputFolderPath = settingsService.Settings.OutputFolderPath;
+
+        if (!string.IsNullOrEmpty(outputFolderPath))
+        {
+            PatchList = patchRepository.LoadPatchesToList(outputFolderPath);
+        }
     }
+    
+    public List<PatchInfo> PatchList { get; set; }  = new();
+    public string RepoFolderPath { get; set; } = string.Empty;
 
-    public void ApplyPatch()
+    public void ExportPatch(string sha, string commitName)
     {
-        throw new NotImplementedException();
+        PatchList.Add(_patchRepository.CreatePatchFile(sha, commitName));
```

Comment:
I also feel adding the patch to the list of imported patches is not the right choice here. 
We should (if we want to display the patches which were creted) use and keep two lists for that. 
That would then also remove the patchCount variable as we can then just use the list of created patches and count the number of objects in that.

Reply:
I actually meant to take that out and forgot. I think just reloading the folder on patch creation makes more sense than having to keep track of two lists that essentially hold the same data